### PR TITLE
Fix show items in package creation grid when shipping is already created for Online Shipping Carriers

### LINKED ISSUE #34964: Can't create label for existing shipment
### Preconditions (*)

1. Have at least one online carrier configured

### Steps to reproduce (*)

1. Create a shipment without selecting the `Create Shipping Label` checkbox. 
2. Try to create a shipping label after the shipment is created.

### Expected result (*)

1. In the pop-up you should be capable to distribute the items to packages

### Actual result (*)

1. Even though the ajax (GetShippingItemsGrid) requests returns the order items, no products are displayed in the pop-up.

The issue seems to be in `Magento/Shipping/view/adminhtml/web/order/packaging.js` in `processPackagePrepare` function, where the quantity is retrieved from input text. When there is a shipment, the input text does not exits. 

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by johnulecan**
### Description (*)

When the shipping is already created, the package editor still checks for qty field of the shipped items to validate them, but if the shipping was previously created, the qty field for items is missing.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#34964

### Manual testing scenarios (*)

1. Setup an Online Shipping Carrier
2. Create an order selecting the online shipping method
3. Create the shipment without checking the checkbox 'Create shipping label'
4. Edit the shipment
5. Create Shipment Label
6. Try to add products to package

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run Functional Tests CE,Functional Tests EE,Functional Tests B2B

**Comment by johnulecan** [comment]
@magento run Functional Tests CE,Functional Tests EE

**Comment by johnulecan** [comment]
@magento run Functional Tests CE

**Comment by Den4ik** [comment]
@magento run Functional Tests CE

**Comment by johnulecan** [comment]
@magento run Functional Tests CE

**Comment by johnulecan** [comment]
Hey @Den4ik The Area: Frontend label on this PR is wrong. It should be Adminhtml.

**Comment by johnulecan** [comment]
ping @Den4ik

File: app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml [hunk 1]
```diff
@@ -13,15 +13,20 @@
...
+        <?php $_i = 0; foreach ($_items as $_item):
+            if (!empty($_item->getOrderItem())):
+                if ($_item->getOrderItem()->getParentItem()):
                     continue;
                 endif;
                 $_i++ ?>
                 <tbody class="<?= /* @noEscape */ $_i%2 ? 'odd' : 'even' ?>">
                     <?= $block->getItemHtml($_item) ?>
                     <?= $block->getItemExtraInfoHtml($_item->getOrderItem()) ?>
+                    <?php $item = $_item->getOrderItem(); ?>
```
Comment (Den4ik) [inline app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml]: Please move it before `$block->getItemExtraInfoHtml($_item->getOrderItem())` and replace this line with `$block->getItemExtraInfoHtml($item)`
It's prevent twice call for `getOrderItem` method

**Review by Den4ik — CHANGES_REQUESTED** [review]
@johnulecan Thanks for PR and sorry for later update.
Please look at my comment

**Comment by johnulecan** [comment]
@magento run all tests

File: app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml (hunk 1 above)
Comment (johnulecan) [reply app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml]: hey @Den4ik . Changes are applied.

**Comment by johnulecan** [comment]
@magento run Functional Tests CE

File: app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml [hunk 2]
```diff
@@ -13,15 +13,22 @@
...
                     <?= $block->getItemHtml($_item) ?>
-                    <?= $block->getItemExtraInfoHtml($_item->getOrderItem()) ?>
+                    <?php $orderItem = $_item->getOrderItem(); ?>
+                    <?= $block->getItemExtraInfoHtml($orderItem) ?>
+                    <?php $qtyToShip = $orderItem->getQtyShipped() ??
+                        (float) $orderItem->getQtyOrdered() - (float) $orderItem->getQtyShipped(); ?>
+                    <?php if ($orderItem->getQtyShipped()): ?>
+                        <input type="hidden" name="shipment[items][<?= /** @noEscape */ $_item->getId()?>]"
+                               value="<?= /** @noEscape */ $qtyToShip; ?>">
+                    <?php endif; ?>
```
Comment (Den4ik) [inline app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml:31]: I see that value from this input will be overridden to correct value from shipment, but proposed solution is confusing.
I believe that for hidden input value should be used `$_item->getQty()`

**Review by Den4ik — CHANGES_REQUESTED** [review]
@johnulecan Thanks for your work on this fix.
Please look at my comment

File: app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml (hunk 2 above)
Comment (johnulecan) [reply app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml:31]: > I see that value from this input will be overridden to correct value from shipment, but proposed solution is confusing. I believe that for hidden input value should be used `$_item->getQty()`

the issue is that if there is a partial shipment created on the order, we have to pass the remaining quatity to ship, not the entire item qty. if we use ```$_item->getQty()``` the entire order item qty will be passed, even if some of the qty have already been shipped.

Comment (Den4ik) [reply app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml:31]: @johnulecan I'm not agree with you. For exist shipment label should be created for items from current shipment.
Also you may check https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Shipping/view/adminhtml/web/order/packaging.js#L571. Value from input will be replaced by item qty value if value from input is greater

**Review by Den4ik — CHANGES_REQUESTED** [review]

**Comment by ishakhsuvarov** [comment]
Closing this for now as code review has not been passed and comments were not addressed for extended amount of time.

Please feel free to reopen and continue working on this anytime.

**Pull request closed without merging**